cmd/pdatagen/internal/base_slices.go

```diff
@@ -0,0 +1,619 @@
+	assert.EqualValues(t, expectedSlice.Len(), dest.Len())
+
+	// Test MoveAndAppendTo not empty slice
+	generateTest${structName}().MoveAndAppendTo(dest)
+	assert.EqualValues(t, 2*expectedSlice.Len(), dest.Len())
+	for i := 0; i < expectedSlice.Len(); i++ {
+		assert.EqualValues(t, expectedSlice.At(i), dest.At(i))
+		assert.EqualValues(t, expectedSlice.At(i), dest.At(i+expectedSlice.Len()))
+	}
+}
+
+func Test${structName}_CopyTo(t *testing.T) {
+	dest := New${structName}()
+	// Test CopyTo to empty
+	New${structName}().CopyTo(dest)
+	assert.EqualValues(t, New${structName}(), dest)
+
+	// Test CopyTo larger slice
+	generateTest${structName}().CopyTo(dest)
+	assert.EqualValues(t, generateTest${structName}(), dest)
+
+	// Test CopyTo same size slice
+	generateTest${structName}().CopyTo(dest)
+	assert.EqualValues(t, generateTest${structName}(), dest)
+}
+
+func Test${structName}_Resize(t *testing.T) {
+	es := generateTest${structName}()
+	emptyVal := New${elementName}()
+	emptyVal.InitEmpty()
+	// Test Resize less elements.
+	const resizeSmallLen = 4
+	expectedEs := make(map[*${originName}]bool, resizeSmallLen)
+	for i := 0; i < resizeSmallLen; i++ {
+		expectedEs[es.At(i).orig] = true
+	}
+	assert.Equal(t, resizeSmallLen, len(expectedEs))
+	es.Resize(resizeSmallLen)
+	assert.Equal(t, resizeSmallLen, es.Len())
+	foundEs := make(map[*${originName}]bool, resizeSmallLen)
+	for i := 0; i < es.Len(); i++ {
+		foundEs[es.At(i).orig] = true
+	}
+	assert.EqualValues(t, expectedEs, foundEs)
+
+	// Test Resize more elements.
+	const resizeLargeLen = 7
+	oldLen := es.Len()
+	expectedEs = make(map[*${originName}]bool, oldLen)
+	for i := 0; i < oldLen; i++ {
+		expectedEs[es.At(i).orig] = true
+	}
+	assert.Equal(t, oldLen, len(expectedEs))
+	es.Resize(resizeLargeLen)
+	assert.Equal(t, resizeLargeLen, es.Len())
+	foundEs = make(map[*${originName}]bool, oldLen)
+	for i := 0; i < oldLen; i++ {
+		foundEs[es.At(i).orig] = true
+	}
+	assert.EqualValues(t, expectedEs, foundEs)
+	for i := oldLen; i < resizeLargeLen; i++ {
+		assert.EqualValues(t, emptyVal, es.At(i))
+	}
+
+	// Test Resize 0 elements.
+	es.Resize(0)
+	assert.Equal(t, 0, es.Len())
+}
+
+func Test${structName}_Append(t *testing.T) {
+	es := generateTest${structName}()
+	emptyVal := New${elementName}()
+	emptyVal.InitEmpty()
+
+	es.Append(emptyVal)
+	assert.EqualValues(t, *(es.At(7)).orig, *emptyVal.orig)
+
+	emptyVal2 := New${elementName}()
+	emptyVal2.InitEmpty()
+
+	es.Append(emptyVal2)
+	assert.EqualValues(t, *(es.At(8)).orig, *emptyVal2.orig)
+
+	assert.Equal(t, 9, es.Len())
+}`
+
+const sliceValueGenerateTest = `func generateTest${structName}() ${structName} {
+	tv := New${structName}()
+	fillTest${structName}(tv)
+	return tv
+}
+
+func fillTest${structName}(tv ${structName}) {
+	tv.Resize(7)
+	for i := 0; i < tv.Len(); i++ {
+		fillTest${elementName}(tv.At(i))
+	}
+}`
+
+type baseSlice interface {
+	getName() string
+}
+
+// Will generate code only for the slice struct.
```

Comment:
It would be useful to add a comment to explain that this structure is used for describing slices of pointers to structures and also mention that the other alternate is sliceValueStruct and how the 2 are different.

Reply:
Also, maybe name all relevant types and constants `sliceOfPtrs*` and `sliceOfValues*` to make it ever more clear.

Reply:
Done.